### PR TITLE
Object to array with nested expand

### PR DESCRIPTION
**Pull request by mdmunir**
| Q | A |
| --- | --- |
| Is bugfix? | no |
| New feature? | yes |
| Breaks BC? | yes |
| Tests pass? | - |
| Fixed issues | #11476 |

merge `$fields` and `$expand` into just `$fields`. I split from another PR

### CODE REVIEW

File: framework/base/Arrayable.php [hunk 1]
```diff
@@ -82,11 +82,11 @@ public function extraFields();
      *
      * @param array $fields the fields that the output array should contain. Fields not specified
      * in [[fields()]] will be ignored. If this parameter is empty, all fields as specified in [[fields()]] will be returned.
-     * @param array $expand the additional fields that the output array should contain.
+     * @param array $except the additional fields that the output array should contain.
```
Comment (samdark) [inline framework/base/Arrayable.php]: I don't think this name change is needed. These are still list of fields, not exceptions.

**Comment by mdmunir** [comment]
@samdark 
How to define exception fields?

**Comment by samdark** [comment]
As I can see, logic of this field wasn't changed much while name was changed and I wonder if it should be changed. That's it.

**Comment by mdmunir** [comment]
No. parameter `$fields` and `$expand` are merge to just `$fields` then introduce new parameter `$except`.
```php
// previous version
$model->toArray([],['customer', 'product']);

// current version
$model->toArray(['*', 'customer', 'product']);
```

**Comment by Faryshta** [comment]
@samdark the problem with the `$expand` parameter is that it is not intuitive now to call certain fields inside an extra field... and if we can call fields inside the `$expand` parameter then we don't need the `$fields` parameter at all since we can just use `$expand`.

We can just use the `fields()` and `extraFields()` in the model to convert to determine when a parameter can be expanded or is a final field

**Pull request closed without merging**